src/ceph_mds.cc

```diff
@@ -101,28 +96,30 @@ int main(int argc, const char **argv)
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
 
-  auto cct = global_init(NULL, args,
+  auto cct = global_init(nullptr, args,
 			 CEPH_ENTITY_TYPE_MDS, CODE_ENVIRONMENT_DAEMON,
 			 0, "mds_data");
   ceph_heap_profiler_init();
 
   std::string val, action;
-  for (std::vector<const char*>::iterator i = args.begin(); i != args.end(); ) {
+  auto i = args.begin();
+  auto end = args.end();
+  for (; i != end; ) {
```

Comment:
why not for (auto i : args)

Reply:
@ukernel `auto i : args` doesn't work because `ceph_argparse_double_dash` expects iterator to be passed.

Reply:
why not 'for (auto i= args.begin();  i != args.end(); )'

Reply:
Agreed with @ukernel, it has the benefit of limiting the scope of `i` and `end`.